tests/func/test_ls.py

```diff
@@ -0,0 +1,105 @@
+import shutil
+import os
+
+from dvc.repo import Repo
+
+FS_STRUCTURE = {
+    "file1": "",
+    "dir1/file2": "",
+    "dir1/file3": "",
+    "dir2": {},
+    ".gitignore": "",
+}
+
+DVC_STRUCTURE = {"dvc_file1": "content", "dir1/dvc_file2": "content"}
+
+
+def test_ls_repo(tmp_dir, dvc, scm):
+    tmp_dir.scm_gen(FS_STRUCTURE, commit="init")
+    tmp_dir.dvc_gen(DVC_STRUCTURE, commit="dvc")
+
+    files = Repo.ls(tmp_dir)
```

Comment:
We could make ls work with `Path` objects, but it is not worth it and breaks the symmetry with the rest of the commands. So just make sure to covert properly in your tests (they are failing on 3.5 for this reason)

```suggestion
    files = Repo.ls(fspath(tmp_dir))
```